cmd/juju/application/refresher/refresher.go

```diff
@@ -383,9 +384,23 @@ func (r *charmHubRefresher) Allowed(cfg RefresherConfig) (bool, error) {
 			return false, errors.Trace(err)
 		}
 
-		if charm.CharmHub.Matches(curl.Schema) {
+		if !charm.CharmHub.Matches(curl.Schema) {
+			return false, nil
+		}
+
+		if !cfg.Switch {
 			return true, nil
 		}
+
+		if err := r.charmAdder.CheckCharmPlacement(cfg.ApplicationName, curl); err != nil {
+			// If force is used then ignore the error, the user seems to know
+			// what they're doing.
+			if !cfg.Force {
```

Comment:
Should we emit a warning here? Forcing such a placement will probably break the deployment so we should at least warn that we proceed even though the charm cannot be legally placed.